tests/components/mqtt/test_discovery.py

```diff
@@ -374,10 +361,8 @@
 
     async_fire_mqtt_message(hass, "homeassistant/sensor/bla/config", data)
     await hass.async_block_till_done()
-    assert (
-        "implicit state_topic is deprecated, add "
-        '"state_topic":"homeassistant/sensor/bla/state"' in caplog.text
-    )
+    assert ("implicit state_topic is deprecated, add "
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/app/foreign-sub/home-assistant/pullRequest?prid=4636028)